gemrb/plugins/SDLVideo/SDL20Video.cpp

```diff
@@ -245,6 +272,26 @@ int SDL20VideoDriver::CreateSDLDisplay(const char* title, bool vsync)
 	return GEM_OK;
 }
 
+void SDL20VideoDriver::CalculateCustomFullscreen(const SDL_DisplayMode *mode) {
+	// Only cover the case when screen is larger than the actual game window
+	if (!(mode->w >= screenSize.w && mode->h >= screenSize.h)) {
+		return;
+	}
+
+	// Only cover the case when width and height have the same alignment in both cases
+	if (!(mode->w/mode->h > 0 && screenSize.w/screenSize.h > 0)) {
+		return;
+	}
+
+	// e. g. display is 1920x1080, game is 1024x768 -> fullscreen window 1365x768
+	float displayRatio = mode->w * 1.0 / mode->h;
+	auto width = static_cast<int>(floorf(screenSize.h * displayRatio));
```

Comment:
Won't this be problematic on windows? Last time it was `ceilf`. Perhaps use the replacement from there and just adjust in the other direction.

Reply:
You mean what I did to fix by e73f53eeccbf3d525f770709b1407ec8e92f3e4d? The math is a little more here since I don't have the display ratio fix, i. e. what to use for `div`. I'll have look for a general approach.